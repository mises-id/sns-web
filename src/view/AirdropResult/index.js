/*
 * @Author: lmk
 * @Date: 2021-07-16 00:15:24
 * @LastEditTime: 2022-06-22 11:46:09
 * @LastEditors: lmk
 * @Description: createPosts page
 */
import Navbar from "@/components/NavBar";
import React, { useEffect, useState } from "react";
import { useTranslation } from "react-i18next";
import "./index.scss";
import { getAirdropInfo, getAirdropReceive, getTwitterAuth } from "@/api/user";
import { useLocation } from "react-router-dom";
import { useHistory } from "react-router-dom/cjs/react-router-dom.min";
import { TextArea, Toast,Button } from "antd-mobile";
import { shortenAddress } from "@/utils";
import { useSelector } from "react-redux";
const AirdropResult = () => {
  const { t } = useTranslation();
  const location = useLocation();
  const  [status, setStatus] = useState(''); // success, fail , unauth
  const [airdropInfo,setAirdropInfo] = useState({})
  useEffect(() => {
    const search = new URLSearchParams(location.search);
    // code : 0 1 2
    if(search.get('code')!=='2'){
      getAirdropInfo().then(res=>{
        setAirdropInfo(res.twitter || {
          username:search.username,
          misesid:search.misesid,
        })
        setStatus(search.get('code')==='1' ? 'fail' : res.twitter.amount>0 ? 'success' : 'timeFail')
      })
    }
    if(search.get('code')==='2'){
      setStatus('unauth')
    }
    // eslint-disable-next-line
  }, [])
  const history = useHistory()
  const customBack = () => {
    history.replace('/home/me')
  }
  const [getValue, setValue] = useState('')
  const [loading,setLoading] = useState(false)
  const getAirdrop = () => {
    
    setLoading(true)
    getAirdropReceive({
      tweet: `${getValue}\n\n${unEditText.replace(/<br\/>/g, '\n')}`
    }).then(res=>{
      Toast.show({
        content: 'Send Success',
        duration: 1500,
        afterClose: () => {
          const coin = airdropInfo?.amount
          history.replace(`/airdrop?isFrom=homePage&MIS=${coin}`)
        }
      })
    }).finally(()=>{
      setLoading(false)
    })
  }
  const getAuth = () => {
    setLoading(true)
    getTwitterAuth()
      .then((res) => {
        window.location.href = res.url;
      })
      .catch((err) => {
        console.log(err);
      }).finally(()=>{
        setLoading(false)
      });
  };
  const statusTxt = ()=>{
    if(status==='fail') return 'This Account has been verified'
    if(status==='timeFail') return 'This Account was created after May. 1, 2022'
    return 'Sorry, you canceled the authorization!'
  }
  const selector = useSelector(state => state.user) || {};
  const misesid = airdropInfo?.misesid || selector.loginForm?.misesid.replace('did:mises:','')
  const unEditText = `I have claimed ${airdropInfo.amount}$MIS airdrop, come and join #Mises to experience the coolest decentralized social media with me!
  <br/><br/>https://www.mises.site/download?MisesID=${misesid}<br/><br/>#Mises #Decentralized #SocialMedia`
  return (
    <>
      <Navbar title={t('airdropPageTitle')} customBack={customBack}/>
      <div className="m-layout airdrop-content" style={{backgroundImage:'url(/static/images/airdrop-bg.png)'}}>
        <div className="airdrop-box">
          <div className="listItem">
            <span className="label">Mises ID:</span>
            <span className="value">{shortenAddress(misesid)}</span>
          </div>
          <div className="listItem">
            <span className="label">Twitter ID:</span>
            <span className="value">{airdropInfo.username || 'NULL'}</span>
          </div>

          {status==='success'&&<>
            <div className="listItem">
              <span className="label">Available Airdrop:</span>
              <span className="value">{airdropInfo.amount}MIS</span>
            </div>
            <p className="text-bold success-tips">Now send this Tweet to get airdrop!</p>
            <div className="text-area">
              <TextArea 
                value={getValue}
                autoSize={{ minRows: 0, maxRows: 4 }}
                style={{'--font-size':'14px',marginBottom:'10px'}}
                onChange={setValue}
                placeholder="Please enter your tweet here"
              />
              <p className="font-14" dangerouslySetInnerHTML={{__html:unEditText}}></p>
            </div>
            <Button 
              className="btn" 
              fill='solid' 
              color='primary'
              loading={loading}
              onClick={getAirdrop}>
              <span>Send Tweet and get Airdrop</span>
            </Button>
          </>}

          {status!=='success'&&status!==''&&<>
            <p className="fail-reason">Your account does not meet the requirements</p>
            <p className="fail-reason">
              Reason: {statusTxt()}
            </p>
            <Button 
              className="btn-fail" 
              fill='outline' 
              color='primary'
              loading={loading}
              onClick={getAuth}>
              <span>Change Twitter</span>
            </Button>
          </>}
        </div>
      </div>
    </>
  );
};
export default AirdropResult;
