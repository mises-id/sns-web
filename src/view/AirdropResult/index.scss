.waiting{
  font-size: 18px;
  font-weight: bold;
  color: #333;
  line-height: 25px;
  text-align: center;
}
.criterion{
  color: #999999;
  font-size: 12px;
  padding: 25px 0;
  line-height: 1;
  text-align: center;
}
.success-img{
  display: block;
  margin: 50px auto 0;
}
.viewInWallet{
  width: 180px;
  margin: 20px auto;
}
.airdrop-content{
  background-size: 100% 179.2vw;
}
.airdrop-box{
  background: white;
  margin: 90px 15px 30px;
  background: #FFFFFF;
  border: 3px solid #BAEBFC;
  border-radius: 15px;
  padding: 16px;
}
.listItem{
  display: flex;
  font-size: 16px;
  margin-bottom: 15px;
  .label{
    color: #666666;
  }
  .value{
    color: #333;
  }
}
.success-tips{
  font-size: 16px;
  font-weight: bold;
  color: #333333;
  margin-top: 10px;
  line-height:20px;
  margin-bottom: 10px;
  word-break: break-word;
}
.twitter-tips{
  font-size: 14px;
  font-weight: bold;
  color: #333;
  margin-bottom: 10px;
  line-height:18px;
  word-break: keep-all;
}
.success-misesid{
  background: #F8F8F8;
  border: 1px solid #EEEEEE;
  border-radius: 5px;
  padding: 14px;
  color: #333333;
  font-size: 14px;
}
.misesid-title{
  color: #666666;
  display: inline-block;
  margin-right: 5px;
}
.btn{
  height: 50px;
  border-radius: 50px;
  width: 100%;
  margin: 25px 0;
  background: #FFD72F;
  color:#000;
  font-size: 17px;
  font-weight: bold;
  border: none;
  box-shadow: 0px 8px 15px 0px rgba(255, 215, 47, 0.3);
}
.btn-fail{
  height: 50px;
  border-radius: 50px;
  width: 100%;
  color:#000;
  border-color: #000;
  margin: 10px 0 25px;
}
.fail-reason{
  font-size: 14px;
  font-weight: 400;
  color: #FF6464;
  margin-bottom: 15px;
  word-break: break-word;
}
.get-airdrop-reason{
  font-size: 14px;
  font-weight: 400;
  color: rgb(65, 160, 65);
  margin-bottom: 15px;
  word-break: break-word;
}
.font-14{
  font-size: 14px;
}
.airdrop-tips{
  word-break: break-word;
  color: #999999;
  text-align: center;
  margin-top: 15px;
  line-height: 18px;
}
.hourglass{
  width: 35px;
  display: block;
  margin: 16px auto;
}
.remove-twitter-auth{
  color: #5D61FF;
  font-size: 12px;
  text-align: center;
  display: block;
  margin-top: 20px;
}