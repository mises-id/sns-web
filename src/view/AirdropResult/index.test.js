/*
 * @Author: lmk
 * @Date: 2021-07-16 10:05:53
 * @LastEditTime: 2022-06-17 16:01:13
 * @LastEditors: lmk
 * @Description: AirdropSuccess test
 */

import React from 'react';
import {shallow } from 'enzyme';
import AirdropResult from '.';
describe('GreatePosts', () => {
  it('should render correctly', () => {
    const component = shallow(< AirdropResult/>)
    expect(component).toMatchSnapshot();
  });
});