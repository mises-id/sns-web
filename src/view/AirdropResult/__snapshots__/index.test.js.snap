// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GreatePosts should render correctly 1`] = `
<Fragment>
  <Navbar
    customBack={[Function]}
    title="airdropPageTitle"
  />
  <div
    className="m-layout airdrop-content"
    style={
      Object {
        "backgroundImage": "url(/static/images/airdrop-bg.png)",
      }
    }
  >
    <div
      className="airdrop-box"
    >
      <div
        className="listItem"
      >
        <span
          className="label"
        >
          Mises ID:
        </span>
        <span
          className="value"
        />
      </div>
      <div
        className="listItem"
      >
        <span
          className="label"
        >
          Twitter:
        </span>
        <span
          className="value"
        >
          NULL
        </span>
      </div>
      <div
        className="listItem"
      >
        <span
          className="label"
        >
          Airdrop:
        </span>
        <span
          className="value"
        >
          MIS
        </span>
      </div>
      <p
        className="text-bold success-tips"
      >
        Now send this Tweet to get airdrop!
      </p>
      <div
        className="text-area"
      >
        <ForwardRef
          autoSize={
            Object {
              "maxRows": 4,
              "minRows": 1,
            }
          }
          className="font-14"
          defaultValue=""
          onChange={[Function]}
          rows={2}
          showCount={false}
          value="Hello everyone, I am Alice, an art lover, this is my personal website welcome to visit."
        />
        <p
          className="font-14"
        >
          I have claimed undefined$MIS airdrop, come and join #Mises to experience the coolest decentralized social media with me!
  Join us and 3% airdrop!
  https://www.mises.site/download?MisesID=undefined
  #Mises #Decentralized #SocialMedia
        </p>
      </div>
      <ForwardRef
        className="btn"
        color="primary"
        fill="solid"
        loading={false}
        onClick={[Function]}
      >
        <span>
          Send Tweet and get Airdrop
        </span>
      </ForwardRef>
    </div>
  </div>
</Fragment>
`;
